app/Http/Controllers/Api/ChannelController.php

```diff
@@ -0,0 +1,359 @@
+    {
+        $userId = 1;// The id of the user gotten from the token
+
+        if ($this->validateUserRequestForEmptyFields($request)) {
+            return Response::json(['message' => 'All fields are required'], 400);
+
+        }
+
+        if ($this->validateUserRequest($request)) {
+            return Response::json(['message' => 'Channel already exists'], 400);
+
+        } 
+
+        $channel = Channel::create([
+            'channel_name' => strtolower($request->input('name')),
+            'channel_description' => $request->input('description'),
+            'created_at' => date('Y-m-d h:i:s'),
+            'user_id' => $userId,
+        ]);
+
+        if (count($channel) > 0) {
+            return Response::json(['message' => 'Channel created successfully'], 201);
+        }
+    }
+
+    /**
+     * This method updates a channel using PUT verb
+     *
+     * @param $channel_name
+     * @param $channel_description
+     *
+     * @return $response
+     */
+    public function editAChannel(Request $request, $channel_name)
+    {
+        if ($this->validateUserRequestForEmptyFields($request)) {
+            return Response::json(['message' => 'All fields are required'], 400);
+
+        }
+
+        $channel = Channel::where('channel_name', '=', strtolower(urldecode($channel_name)))
+        ->first();
+
+        if ($this->processEditChannel($request, $channel)) {
+            return Response::json(['message' => 'Channel updated successfully'], 200);
+
+        }
+
+        return Response::json([
+            'message' => 'Channel cannot be updated because the channel name is incorrect'
+        ], 404);
+
+    }
+
+    /**
+     * This method updates a channel using PATCH verb
+     *
+     * @param $channel_name
+     *
+     * @return $response
+     */
+    public function editASingleChannelResource(Request $request, $channel_name)
+    {
+        if ($this->validateUserRequestForPatchRequest($request)) {
+            return Response::json(['message' => 'All fields are required'], 400);
+
+        }
+
+        $channel = Channel::where('channel_name', '=', strtolower(urldecode($channel_name)))
+        ->first();
+
+        if ($this->processEditChannelForPatchRequest($request, $channel)) {
+            return Response::json(['message' => 'Channel updated successfully'], 200);
+
+        }
+
+        return Response::json([
+            'message' => 'Channel cannot be updated because the channel name is incorrect'
+        ], 404);
+
+    }
+
+    /**
+     * This method deletes a channel
+     *
+     * @param $channel_name
+     *
+     * @param $request
+     *
+     * @return $response
+     */
+    public function deleteASingleChannel(Request $request, $channel_name)
+    {
+        $channel = Channel::where('channel_name', '=', strtolower(urldecode($channel_name)))
+        ->first();
+
+        if (! is_null($channel)) {
+            $returnValue = $this->channelRepository->deleteChannel($channel->id);
```

Comment:
**Code Quality- Nitpick**
@andela-tolotin There is possibility that something goes wrong and the delete will not work, Its better if you do a check here to be sure that the soft delete worked before sending out the message.